### PR TITLE
add Transaction#clearOutputs

### PR DESCRIPTION
**Pull request by maraoz**
This PR adds a `clearOutputs()` method for `Transaction` that removes all outputs.

### CODE REVIEW

File: lib/transaction/transaction.js [hunk 1]
```diff
@@ -713,6 +713,21 @@ Transaction.prototype.addOutput = function(output) {
...
+ * Remove all outputs from the transaction.
+ *
+ * @return {Transaction} this, for chaining
+ */
+Transaction.prototype.clearOutputs = function() {
+  this.outputs = [];
+  this._clearSignatures();
+  this._outputAmount = undefined;
+  this._changeIndex = undefined;
+  return this;
```
Comment (eordano) [inline lib/transaction/transaction.js]: Shouldn't this also call `this._updateChangeOutput()`? Otherwise all the inputs will be miner's fee.

**Comment by maraoz** [comment]
updated

**Comment by braydonf** [comment]
LGTM

**Comment by pnagurny** [comment]
ACK

**Comment by maraoz** [comment]
Yay!

**Pull request merged**